### PR TITLE
sync : ggml

### PR DESCRIPTION
**Pull request by ggerganov**

### CODE REVIEW

**Comment by ggerganov** [comment]
@JohannesGaessler @slaren How to fix the HIP build:

```txt
[ 11%] Building HIP object ggml/src/CMakeFiles/ggml.dir/ggml-cuda/out-prod.cu.o
In file included from /__w/llama.cpp/llama.cpp/ggml/src/ggml-cuda/out-prod.cu:2:
/__w/llama.cpp/llama.cpp/ggml/src/ggml-cuda/vendors/cuda.h:3:10: fatal error: 'cuda_runtime.h' file not found
#include <cuda_runtime.h>
         ^~~~~~~~~~~~~~~~
1 error generated when compiling for gfx906.
gmake[2]: *** [ggml/src/CMakeFiles/ggml.dir/build.make:404: ggml/src/CMakeFiles/ggml.dir/ggml-cuda/out-prod.cu.o] Error 1
```

File: ggml/src/ggml-cuda/out-prod.cu [hunk 1]
```diff
@@ -0,0 +1,52 @@
+#include "out-prod.cuh"
+#include "vendors/cuda.h"
```
Comment (slaren) [inline ggml/src/ggml-cuda/out-prod.cu:2]:
```suggestion
```

Comment (slaren) [reply ggml/src/ggml-cuda/out-prod.cu:2]: This should do it, this file cannot be included directly, and it is already included in `common.cuh`.

Comment (JohannesGaessler) [reply ggml/src/ggml-cuda/out-prod.cu:2]: I can confirm that this fixes compilation with `GGML_HIPBLAS`.

**Pull request merged**